### PR TITLE
DEV-159 add pre-commit and detect-secrets

### PR DESCRIPTION
**Pull request by asurinsaka**
We would like to use pre-commit to perform various checks for us. 
Detecting secrets before committing is one use-case.

### CODE REVIEW

File: test-reports/results.xml [hunk 1]
```diff
@@ -0,0 +1 @@
...st_es_models.py" line="25" name="test_get_es_models_standard_behavior" time="5.938"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="43" name="test_get_es_models_parametrized[mock_listdir0]" time="0.001"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="43" name="test_get_es_models_parametrized[mock_listdir1]" time="0.001"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="57" name="test_get_es_models_from_directory" time="0.021"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="70" name="test_get_models_with_descriptions" time="0.007"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="80" name="test_get_models_no_descriptions" time="0.005"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="90" name="test_get_models_empty_descriptions" time="0.005"></testcase><testcase classname="tests.test_es_models" file="tests/test_es_models.py" line="100" name="test_get_models_descriptions_with_other_data" time="0.006"></testcase></testsuite></testsuites>
```
Comment (anmaxvl) [inline test-reports/results.xml:1]: I don't think we need this.

File: dev-requirements.txt [hunk 2]
```diff
@@ -2,27 +2,30 @@
 # This file is autogenerated by pip-compile
 # To update, run:
 #
-#    pip-compile --output-file=dev-requirements.txt dev-requirements.in setup.py
```
Comment (philtom-ctds) [inline dev-requirements.txt:5]: Looks like `setup.py` was omitted when compiling these dev requirements.   See the section on pinning dev dependencies.   This repo is a library.
https://wiki.uchicago.edu/display/CDIS/GDC+Dependency+Management#GDCDependencyManagement-PinningDevDependencies

File: test-reports/results.xml (hunk 1 above)
Comment (philtom-ctds) [reply test-reports/results.xml:1]: Yes, this file should be omitted.

Comment (asurinsaka) [reply test-reports/results.xml:1]: Should I add test-reposts/ to .gitignore?

**Pull request merged**